Anvil doesn't compile on Windows due to IPC

### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

42e7153b0a7aadb779f97b2c0675bc2206f774d6

### What command(s) is the bug in?

cargo install

### Operating System

Windows

### Describe the bug

On Windows, installing Anvil with `cargo install` fails due to the absence of IPC support in ethers:

```
$ cargo install --git https://github.com/foundry-rs/foundry --locked anvil
...
   Compiling anvil-core v0.1.0 (C:\Users\Michael\.cargo\git\checkouts\foundry-f7cca724e93059b0\42e7153\anvil\core)
error[E0412]: cannot find type `Ipc` in crate `ethers::providers`
   --> anvil\src\lib.rs:269:76
    |
269 |     pub async fn ipc_provider(&self) -> Option<Provider<ethers::providers::Ipc>> {
    |                                                                            ^^^ not found in `ethers::providers`

error[E0599]: no function or associated item named `connect_ipc` found for struct `ethers::ethers_providers::Provider` in the current scope
   --> anvil\src\lib.rs:272:34
    |
272 |         let provider = Provider::connect_ipc(&ipc_path).await.unwrap_or_else(|err| {
    |                                  ^^^^^^^^^^^
    |                                  |
    |                                  function or associated item not found in `Provider<_>`
    |                                  help: there is an associated function with a similar name: `connect`

Some errors have detailed explanations: E0412, E0599.
For more information about an error, try `rustc --explain E0412`.
```

This is remedied by enabling the IPC feature like so:

```
$ cargo install --git https://github.com/foundry-rs/foundry --locked anvil --features ethers/ipc
```

It looks like there's logic to switch IPC _off_ on Windows here:

https://github.com/foundry-rs/foundry/blob/42e7153b0a7aadb779f97b2c0675bc2206f774d6/anvil/Cargo.toml#L69-L73

However, I think this prevents compilation on Windows at all, because the code expects IPC support to be available. If a feature flag were added to anvil for IPC then it could be used to enable/disable the IPC support and prevent the error.

--- a/anvil/Cargo.toml
+++ b/anvil/Cargo.toml
@@ -27,7 +27,7 @@ foundry-config = { path = "../config" }
 
 # evm support
 bytes = "1.4.0"
-ethers = { workspace = true, features = ["ws"] }
+ethers = { workspace = true, features = ["rustls", "ws", "ipc"] }
 trie-db = { version = "0.23" }
 hash-db = { version = "0.15" }
 memory-db = { version = "0.29" }
@@ -66,11 +66,6 @@ fdlimit = { version = "0.2", optional = true }
 clap_complete_fig = "4"
 ethereum-forkid = "0.12"
 
-# ethers
-[target.'cfg(not(windows))'.dependencies]
-ethers = { workspace = true, features = ["rustls", "ws", "ipc"] }
-[target.'cfg(windows)'.dependencies]
-ethers = { workspace = true, features = ["rustls", "ws"] }
 
 [dev-dependencies]
 ethers = { workspace = true, features = ["abigen"] }

